src/chainstate/coordinator/tests.rs

```diff
@@ -2842,169 +3009,1101 @@ fn test_pox_processable_block_in_different_pox_forks() {
-    //   10 will process fine, because its parent has consensus hash =
-    //       INITIAL_CONSENSUS_HASH
-    //   11 will NOT process fine, even though it _should_, because its parents
-    //     consensus hash is different than the consensus hash of the parent when it was mined
+    // same canonical Stacks chain tip
+    let stacks_tip = SortitionDB::get_canonical_stacks_chain_tip_hash(sort_db.conn()).unwrap();
+    let stacks_tip_blind =
+        SortitionDB::get_canonical_stacks_chain_tip_hash(sort_db_blind.conn()).unwrap();
+    assert_eq!(stacks_tip, stacks_tip_blind);
 
-    let sort_id = SortitionDB::get_block_snapshot_for_winning_stacks_block(
-        &sort_db_blind.index_conn(),
-        &SortitionDB::get_canonical_sortition_tip(sort_db_blind.conn()).unwrap(),
-        &stacks_blocks[10].1.block_hash(),
-    )
-    .unwrap()
-    .unwrap()
-    .sortition_id;
+    // same final consensus hash, at the start of height 20
+    let tip = SortitionDB::get_canonical_burn_chain_tip(sort_db.conn()).unwrap();
+    let blinded_tip = SortitionDB::get_canonical_burn_chain_tip(sort_db_blind.conn()).unwrap();
 
-    reveal_block(
-        path_blinded,
-        &sort_db_blind,
-        &mut coord_blind,
-        &sort_id,
-        &stacks_blocks[10].1,
+    assert!(tip.sortition);
+    assert!(blinded_tip.sortition);
+    assert_eq!(
+        tip.winning_stacks_block_hash,
+        blinded_tip.winning_stacks_block_hash
     );
+    assert_eq!(tip.burn_header_hash, blinded_tip.burn_header_hash);
+    assert_eq!(tip.consensus_hash, blinded_tip.consensus_hash);
+    assert_eq!(tip.block_height, 21);
+    assert_eq!(blinded_tip.block_height, 21);
+}
 
-    let block_height = eval_at_chain_tip(path_blinded, &sort_db_blind, "block-height");
-    assert_eq!(block_height, Value::UInt(2));
-    eprintln!("Processed block 10 okay!");
-
-    // won't successfully process the block
-    let sort_id = SortitionDB::get_block_snapshot_for_winning_stacks_block(
-        &sort_db_blind.index_conn(),
-        &SortitionDB::get_canonical_sortition_tip(sort_db_blind.conn()).unwrap(),
-        &stacks_blocks[11].1.block_hash(),
-    )
-    .unwrap()
-    .unwrap()
-    .sortition_id;
+fn highest_block(sortitions: &HashMap<u64, BlockSnapshot>) -> BlockSnapshot {
+    let mut max_k = 0;
+    for (k, _) in sortitions.iter() {
+        if *k > max_k {
+            max_k = *k;
+        }
+    }
+    sortitions.get(&max_k).clone().unwrap().to_owned()
+}
 
-    reveal_block(
-        path_blinded,
-        &sort_db_blind,
-        &mut coord_blind,
-        &sort_id,
-        &stacks_blocks[11].1,
+fn replay_reward_cycle<'a>(
+    stacks_blocks: &Vec<(SortitionId, StacksBlock, u64)>,
+    b: &Burnchain,
+    coord: &mut ChainsCoordinator<'a, NullEventDispatcher, (), OnChainRewardSetProvider>,
+    sort_db: &SortitionDB,
+    chainstate: &mut StacksChainState,
+) {
+    for _ in 0..b.pox_constants.reward_cycle_length {
```

Comment:
could this function be used above in `test_pox_processable_block_in_different_pox_forks`? there seems to be a lot of similar code.

Reply:
They're different enough that it's probably not worth the effort.